test/src/module/storage/gcsTest.c

```diff
@@ -0,0 +1,941 @@
+
+    // Content
+    if (param.content != NULL)
+    {
+        strCatFmt(
+            response,
+            "content-length:%zu\r\n"
+                "\r\n"
+                "%s",
+            strlen(param.content), param.content);
+    }
+    else
+        strCatZ(response, "\r\n");
+
+    hrnServerScriptReply(write, response);
+}
+
+/***********************************************************************************************************************************
+Test Run
+***********************************************************************************************************************************/
+void
+testRun(void)
+{
+    FUNCTION_HARNESS_VOID();
+
+    Storage *storageTest = storagePosixNewP(strNew(testPath()), .write = true);
+
+    // Get test host and ports
+    const String *const testHost = hrnServerHost();
+    const unsigned int testPort = hrnServerPort(0);
+    const unsigned int testPortAuth = hrnServerPort(1);
+
+    // *****************************************************************************************************************************
+    if (testBegin("storageRepoGet()"))
+    {
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("storage with default options");
+
+        StringList *argList = strLstNew();
+        strLstAddZ(argList, "--" CFGOPT_STANZA "=test");
+        hrnCfgArgRawZ(argList, cfgOptRepoType, STORAGE_GCS_TYPE);
+        hrnCfgArgRawZ(argList, cfgOptRepoPath, "/repo");
+        hrnCfgArgRawZ(argList, cfgOptRepoGcsBucket, TEST_BUCKET);
+        hrnCfgArgRawZ(argList, cfgOptRepoGcsKeyType, STORAGE_GCS_KEY_TYPE_TOKEN);
+        hrnCfgEnvRawZ(cfgOptRepoGcsKey, TEST_TOKEN);
+        harnessCfgLoad(cfgCmdArchivePush, argList);
+
+        Storage *storage = NULL;
+        TEST_ASSIGN(storage, storageRepoGet(0, false), "get repo storage");
+        TEST_RESULT_STR_Z(storage->path, "/repo", "    check path");
+        TEST_RESULT_STR(((StorageGcs *)storage->driver)->bucket, TEST_BUCKET_STR, "    check bucket");
+        TEST_RESULT_STR_Z(((StorageGcs *)storage->driver)->endpoint, "storage.googleapis.com", "    check endpoint");
+        TEST_RESULT_UINT(((StorageGcs *)storage->driver)->chunkSize, STORAGE_GCS_CHUNKSIZE_DEFAULT, "    check chunk size");
+        TEST_RESULT_STR(((StorageGcs *)storage->driver)->token, TEST_TOKEN_STR, "    check token");
+        TEST_RESULT_BOOL(storageFeature(storage, storageFeaturePath), false, "    check path feature");
+        TEST_RESULT_BOOL(storageFeature(storage, storageFeatureCompress), false, "    check compress feature");
+    }
+
+    // *****************************************************************************************************************************
+    if (testBegin("storageGcsAuth*()"))
+    {
+        StorageGcs *storage = NULL;
+
+        // !!! HACKY WAY TO GET A BEARER TOKEN FOR TESTING AT THE COMMAND LINE
```

Comment:
As discussed, I'd prefer to remove it. Since it is commented out it will rot and I think a THROW in the relevant code section is just as easy. Removed in b6a19f68bd27c8b008.